eng/build.sh

```diff
@@ -2,63 +2,514 @@
+        fi
+    fi
+
+    if [ ! -f $LLDB_LIB ]; then
+        echo "Cannot find the lldb library. Try installing Xcode."
+        exit 1
+    fi
+
+    # Workaround bad python version in /usr/local/bin/python2.7 on lab machines
+    export PATH=/usr/bin:$PATH
+    which python
+    python --version
+fi
+
+#
+# Build native components
+#
+
+if [ ! -e $__DotNetCli ]; then
+   echo "dotnet cli not installed $__DotNetCli"
+   exit 1
+fi
+
+if [ $__NativeBuild == true ]; then
+    if [[ $__CI == true ]]; then
+        echo "Generating Version Source File"
+        __GenerateVersionRestoreLog="$__LogDir/GenerateVersionRestore.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionRestoreLog /t:Restore /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+
+        __GenerateVersionLog="$__LogDir/GenerateVersion.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionLog /t:GenerateVersionFiles /p:GenerateVersionSourceFile=true /p:NativeVersionSourceFile="$__IntermediatesDir/version.cpp" /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+        if [ $? != 0 ]; then
+            echo "Generating Version Source File FAILED"
+            exit 1
+        fi
+    else
+        echo "Generating Empty Version Source File"
+       echo "" > "$__IntermediatesDir/version.cpp"
+    fi
+
+    build_native "$__BuildArch" "$__IntermediatesDir" "$__ExtraCmakeArgs"
+fi
+
+#
+# Copy the native SOS binaries to where these tools expect for testing
+#
+
+if [[ $__NativeBuild == true || $__Test == true ]]; then
+    __dotnet_sos=$__RootBinDir/bin/dotnet-sos/$__BuildType/netcoreapp2.1/publish/$__DistroRid
+    __dotnet_dump=$__RootBinDir/bin/dotnet-dump/$__BuildType/netcoreapp2.1/publish/$__DistroRid
+
+    mkdir -p "$__dotnet_sos"
+    mkdir -p "$__dotnet_dump"
+
+    cp "$__BinDir"/* "$__dotnet_sos"
+    echo "Copied SOS to $__dotnet_sos"
+
+    cp "$__BinDir"/* "$__dotnet_dump"
+    echo "Copied SOS to $__dotnet_dump"
+fi
+
+#
+# Run xunit tests
+#
+
+if [ $__Test == true ]; then
+   if [ $__CrossBuild != true ]; then
+      if [ "$LLDB_PATH" == "" ]; then
+          export LLDB_PATH="$(which lldb-3.9.1 2> /dev/null)"
+          if [ "$LLDB_PATH" == "" ]; then
+              export LLDB_PATH="$(which lldb-3.9 2> /dev/null)"
+              if [ "$LLDB_PATH" == "" ]; then
+                  export LLDB_PATH="$(which lldb-4.0 2> /dev/null)"
+                  if [ "$LLDB_PATH" == "" ]; then
+                      export LLDB_PATH="$(which lldb-5.0 2> /dev/null)"
+                      if [ "$LLDB_PATH" == "" ]; then
+                          export LLDB_PATH="$(which lldb 2> /dev/null)"
+                      fi
+                  fi
+              fi
+          fi
+      fi
+
+      if [ "$GDB_PATH" == "" ]; then
+          export GDB_PATH="$(which gdb 2> /dev/null)"
```

Comment:
What do we use this one for?

Reply:
The SOSRunner uses it to launch gdb. There were times like for .NET Core 1.1 that was how we created dumps to test.  It probably could be cleaned up but not in this PR. It was always there.